src/analysisd/decoders/security_configuration_assessment.c

```diff
@@ -1561,6 +1579,72 @@ static int CheckPoliciesJSON(cJSON *event,cJSON **policies) {
     return retval;
 }
 
+/*
+    Handle integrity check received from the agent. If the integrity hash is different from the
+    stored one or there is no hash stored, it will request a new scan. If the integrity hash is
+    the same one, it will do nothing.
+*/
+static void HandleIntegrityCheck(Eventinfo *lf, int *socket, cJSON *event){
+    assert(lf);
+    assert(event);
+
+    cJSON *integrity_hash = NULL;
+    cJSON *policy_id = NULL;
+
+    integrity_hash = cJSON_GetObjectItem(event, "hash");
+    policy_id =  cJSON_GetObjectItem(event, "policy_id");
+
+    if (!integrity_hash){
+        merror("Malformed JSON: field 'hash' not found.");
+        return;
+    }
+
+    if (!integrity_hash->valuestring) {
+        merror("Malformed JSON: field 'hash' must be a string.");
+        return;
+    }
+
+    if (!policy_id) {
+        merror("Malformed JSON: field 'policy_id' not found.");
+        return;
+    }
+
+    if (!policy_id->valuestring) {
+        merror("Malformed JSON: field 'policy_id' must be a string.");
+        return;
+    }
+
+    char *wdb_response = NULL;
+    os_calloc(OS_MAXSTR, sizeof(char), wdb_response);
+
+    int result_db = FindCheckResults(lf, policy_id->valuestring, socket, wdb_response);
+
+    switch (result_db)  {
+        case 0: /* Integrity check */
+            if(strcmp(wdb_response, integrity_hash->valuestring)) {
```

Comment:
It looks like this condition never occurs. When a new scan determines that the hash has changed, it will automatically send a message to the manager. When the next integrity check occurs the manager will already have stored the new hash